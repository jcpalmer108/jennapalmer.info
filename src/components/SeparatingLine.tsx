import { Line } from '../styles/components/SeparatingLine'

type SeparatingLineProps = {
  thin?: boolean,
  light?: boolean
}

export default function SeparatingLine({ light, thin }: SeparatingLineProps) {
  return (<Line light={light} thin={thin} data-testid="Line" />)
}