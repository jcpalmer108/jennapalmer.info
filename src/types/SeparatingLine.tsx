export type SeparatingLineProps = {
  thin?: boolean,
  light?: boolean
}
