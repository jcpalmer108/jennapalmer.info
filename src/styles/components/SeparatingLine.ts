import styled from "styled-components"
import { silver_polish, tabletBreakpoint } from "../variables"
import { SeparatingLineProps as LineProps } from '../../types/SeparatingLine'

export const Line = styled.hr<LineProps>`
  width: 100%;
  border-color: transparent;
  border-width: 1px;
  border-block-start: none;
  margin: 0px;
  
  @media (min-width: ${tabletBreakpoint}px) {
    border-color: ${props => props.light ? "white" : silver_polish};
    border-width: ${props => props.thin ? 0.5 : 1}px;
    position: absolute;
    left: 0px;
    margin: 0px;
  }

`
