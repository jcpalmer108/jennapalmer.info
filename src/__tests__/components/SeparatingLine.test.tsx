import { render, screen } from "@testing-library/react";
import SeparatingLine from "../../components/SeparatingLine"

describe('SeparatingLine', () => {
  test('Happy Path', () => {
    render(<SeparatingLine light thin />)
    expect(screen.getByTestId("Line")).toMatchSnapshot()
  })

  test('Un-Happy Path', () => {
    render(<SeparatingLine />)
    expect(screen.getByTestId("Line")).toMatchSnapshot()
  })
})
