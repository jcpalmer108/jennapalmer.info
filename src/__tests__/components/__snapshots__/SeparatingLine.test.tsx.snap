// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SeparatingLine renders if optional params are not passed in 1`] = `
<hr
  class="sc-bczRLJ Rnrfi"
  data-testid="Line"
/>
`;

exports[`SeparatingLine renders if optional params are passed in 1`] = `
<hr
  class="sc-bczRLJ cmKAyx"
  data-testid="Line"
/>
`;
